src/test/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/resource/MigrateResourceIntTest.kt

```diff
@@ -0,0 +1,210 @@
+package uk.gov.justice.digital.hmpps.incidentreporting.resource
+
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.TestConfiguration
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Primary
+import uk.gov.justice.digital.hmpps.incidentreporting.helper.buildIncidentReport
+import uk.gov.justice.digital.hmpps.incidentreporting.integration.SqsIntegrationTestBase
+import uk.gov.justice.digital.hmpps.incidentreporting.jpa.IncidentReport
+import uk.gov.justice.digital.hmpps.incidentreporting.jpa.repository.IncidentReportRepository
+import uk.gov.justice.digital.hmpps.incidentreporting.model.nomis.CodeDescription
+import uk.gov.justice.digital.hmpps.incidentreporting.model.nomis.NomisIncidentReport
+import uk.gov.justice.digital.hmpps.incidentreporting.model.nomis.NomisIncidentStatus
+import uk.gov.justice.digital.hmpps.incidentreporting.model.nomis.Offender
+import uk.gov.justice.digital.hmpps.incidentreporting.model.nomis.OffenderParty
+import uk.gov.justice.digital.hmpps.incidentreporting.model.nomis.Staff
+import uk.gov.justice.digital.hmpps.incidentreporting.model.nomis.StaffParty
+import uk.gov.justice.digital.hmpps.incidentreporting.service.InformationSource
+import java.time.Clock
+import java.time.LocalDateTime
+
+private const val INCIDENT_NUMBER: Long = 112414323
+
+class MigrateResourceIntTest : SqsIntegrationTestBase() {
+
+  @TestConfiguration
+  class FixedClockConfig {
+    @Primary
+    @Bean
+    fun fixedClock(): Clock = clock
+  }
+
+  @Autowired
+  lateinit var repository: IncidentReportRepository
+
+  lateinit var existingNomisIncident: IncidentReport
+
+  @BeforeEach
+  fun setUp() {
+    repository.deleteAll()
+
+    existingNomisIncident = repository.save(
+      buildIncidentReport(incidentNumber = "$INCIDENT_NUMBER", reportTime = LocalDateTime.now(clock), source = InformationSource.NOMIS),
```

Comment:
should the incident number here be a string with `IR-0000000112414323` to be more realistic?